Source/Core/DiscIO/VolumeDirectory.cpp

```diff
@@ -505,10 +506,10 @@ static u32 ComputeNameSize(const File::FSTEntry& parent_entry)
   return name_size;
 }
 
-static std::string ASCIIToLowercase(std::string str)
+static std::string ASCIIToUppercase(std::string str)
 {
   std::transform(str.begin(), str.end(), str.begin(),
-                 [](char c) { return std::tolower(c, std::locale::classic()); });
+                 [](char c) { return std::toupper(c, std::locale::classic()); });
```

Comment:
I'm just going off of difference between the real and generated FSTs, but Wiim's ISO Tools [has a "Nintendo" comparison function that mirrors this](https://github.com/mirror/wiimms-iso-tools/blob/8f1b6ecb0cbac3bdc83ceaeb7cff3c9375974485/src/lib-std.c#L1282). The transformation values for a-z are set to the same transformation values for A-Z, which pretty much simplifies to changing the filename to uppercase and comparing it.